### PR TITLE
Feature/connect to sqlalchemy

### PR DESCRIPTION
**Pull request by yybeltranr**
## Background

Connections with SQLAlchemy.

## Changes done

Implementation of requirements to connect with SQLAlchemy. Add complete DB.

## Pending to be done

Improve the write code.

## Notes

Difficulty with paths.

### CODE REVIEW

File: ms-campus-kid-grades/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version: '3.5'
```
Comment (jhonatan247) [inline ms-campus-kid-grades/docker-compose.yml:1]: This file is not necessary for the project architecture, we already have a `docker-compose` file.

File: ms-campus-kid-grades/src/db_postgres/app.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import json
+from flask import Flask
+from flask import request
+import flask_sqlalchemy
+import os
+
+#user = os.environ['postgres']
```
Comment (jhonatan247) [inline ms-campus-kid-grades/src/db_postgres/app.py]: Why do we need these comments?

File: ms-campus-kid-grades/src/db_postgres/app.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+db = flask_sqlalchemy.SQLAlchemy()
+
+class Estudiante(db.Model):
+    __tablename__ = 'estudiante'
+    id = db.Column(db.Integer, primary_key = True)
+    correo_estudiante = db.Column(db.String(100), nullable=False, unique=True)
+    nombre_estudiante = db.Column(db.String(100), nullable=False,)
+    id_facultad = db.Column(db.Integer, nullable=False, unique=True)
+    # created_at = db.Column(db.Datetime(), default=datetime.now())
```
Comment (jhonatan247) [inline ms-campus-kid-grades/src/db_postgres/app.py:17]: Why do we need this comment?

File: ms-campus-kid-grades/src/db_postgres/app.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    flask_app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+    flask_app.app_context().push()
+    db.init_app(flask_app)
+    db.create_all()
+    return flask_app
+
+
+app = create_app()
+
+# Recibir info sobre todos los estudiantes
```
Comment (jhonatan247) [inline ms-campus-kid-grades/src/db_postgres/app.py]: All comments should be written in English.

File: ms-campus-kid-grades/src/db_postgres/app.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+@app.route('/', methods=['GET'])
+def fetch():
+    estudiante = db.get_all(Estudiante)
```
Comment (jhonatan247) [inline ms-campus-kid-grades/src/db_postgres/app.py]: All variables in your code should be written in English.
In your case, all code should be written in English.

File: ms-campus-kid-grades/.gitignore [hunk 6]
```diff
@@ -1 +1,2 @@
-**/__pycache__/
\ No newline at end of file
+**/__pycache__/
+/virtual
```
Comment (jhonatan247) [inline ms-campus-kid-grades/.gitignore]: Please add a new line at the end of the file.

File: ms-campus-kid-grades/database.conf [hunk 7]
```diff
@@ -0,0 +1,6 @@
+POSTGRES_USER= postgres
```
Comment (jhonatan247) [inline ms-campus-kid-grades/database.conf]: These are not the configurations for the database we already have in the docker-compose, please use those ones.
```
DB_HOST = host.docker.internal
DB_USER = root
DB_PASS = 1234
DB_DATABASE = db_grade
DB_PORT = 4006
```
Be sure that your project works well with them.
You have to run the `docker-compose` file.
There are some instructions in the `README.md` file

**Review by jhonatan247 — CHANGES_REQUESTED** [review]
Some comments:
- Please apply all comments in all cases in your code.
- Why did you delete the `Dockerfile`? It is so important and it was already configured.

File: ms-campus-kid-grades/src/db_postgres/app.py (hunk 2 above)
Comment (yybeltranr) [reply ms-campus-kid-grades/src/db_postgres/app.py]: I don't need it.

**Review by jhonatan247 — CHANGES_REQUESTED** [review]
Your branch is not up to date. Please update it with the `dev` branch.

File: ms-campus-kid-grades/src/db_postgres/app.py (hunk 3 above)
Comment (yybeltranr) [reply ms-campus-kid-grades/src/db_postgres/app.py:17]: Lately we can use it in another db to obtain times on insertions.

File: ms-campus-kid-grades/src/db_postgres/app.py (hunk 2 above)
Comment (jhonatan247) [reply ms-campus-kid-grades/src/db_postgres/app.py]: In that case, please delete all of them.

File: ms-campus-kid-grades/src/db_postgres/app.py (hunk 4 above)
Comment (yybeltranr) [reply ms-campus-kid-grades/src/db_postgres/app.py]: hmm ok

File: ms-campus-kid-grades/src/db_postgres/app.py (hunk 5 above)
Comment (yybeltranr) [reply ms-campus-kid-grades/src/db_postgres/app.py]: Ok I'll change it. I just did in this way waiting fot the creation of the db.

File: ms-campus-kid-grades/.gitignore (hunk 6 above)
Comment (yybeltranr) [reply ms-campus-kid-grades/.gitignore]: why?

File: ms-campus-kid-grades/src/db_postgres/app.py [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+            "correo_estudiante": e.correo_estudiante,
+            "nombre_estudiante": e.nombre_estudiante,
+            "id_facultad": e.id_facultad
+        }
+
+        all_estudiante.append(new_estudiante)
+    return json.dumps(all_estudiante), 200
+
+# Añadir nuevo estudiante
+@app.route('/add', methods=['POST'])
```
Comment (jhonatan247) [inline ms-campus-kid-grades/src/db_postgres/app.py:70]: The word `add` is not necessary for the route name
```suggestion
@app.route('/ ', methods=['POST'])
```

File: ms-campus-kid-grades/src/db_postgres/app.py [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+    data = request.get_json()
+    correo_estudiante = data['correo_estudiante']
+    nombre_estudiante = data['nombre_estudiante']
+    id_facultad = data['id_facultad']
+
+    db.add_instance(Estudiante, correo_estudiante=correo_estudiante, nombre_estudiante=nombre_estudiante, id_facultad=id_facultad)
+    return json.dumps("Added"), 200
+
+# Eliminar estudiante
+@app.route('/delete/<id_estudiante>', methods=['DELETE'])
```
Comment (jhonatan247) [inline ms-campus-kid-grades/src/db_postgres/app.py]: The word `delete` is not necessary for the route name
```suggestion
@app.route('/<id_estudiante>', methods=['DELETE'])
```

File: ms-campus-kid-grades/src/db_postgres/app.py [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+def remove(id_estudiante):
+    Estudiante.db.delete_instance(Estudiante.Estudiante, id=id_estudiante)
+    return json.dumps("Deleted"), 200
+
+# Editar estudiante
+@app.route('/edit/<id_estudiante>', methods=['PATCH'])
```
Comment (jhonatan247) [inline ms-campus-kid-grades/src/db_postgres/app.py]: The word `edit` is not necessary for the route name
```suggestion
@app.route('/<id_estudiante>', methods=['PATCH'])
```

**Review by jhonatan247 — CHANGES_REQUESTED** [review]

File: ms-campus-kid-grades/.gitignore (hunk 6 above)
Comment (jhonatan247) [reply ms-campus-kid-grades/.gitignore]: It is required for git that all files have a new line at the end.

File: ms-campus-kid-grades/src/db_postgres/app.py (hunk 8 above)
Comment (yybeltranr) [reply ms-campus-kid-grades/src/db_postgres/app.py:70]: But if you need to add for id, how can you do it?

File: ms-campus-kid-grades/database.conf [hunk 11]
```diff
@@ -1,6 +1,5 @@
-SECRET_KEY = 1234P
 DB_HOST = host.docker.internal
 DB_USER = root
 DB_PASS = 1234
 DB_DATABASE = db_grade
-DB_PORT = 4006
+DB_PORT = 4006
```
Comment (jhonatan247) [inline ms-campus-kid-grades/database.conf:5]: Please add a new line at the end of the file.

**Review by jhonatan247 — CHANGES_REQUESTED** [review]
Just one small comment

**Pull request closed without merging**